src/components/HomePage/ReadingStreak/index.tsx

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable react/no-multi-comp */
+import { useMemo } from 'react';
+
+import Trans from 'next-translate/Trans';
+import useTranslation from 'next-translate/useTranslation';
+
+import CurrentWeekProgress from './CurrentWeekProgress';
+import DeleteReadingGoalButton from './DeleteReadingGoalButton';
+import ReadingGoalModal from './ReadingGoalModal';
+import styles from './ReadingStreak.module.scss';
+
+import VerseLink from '@/components/Verse/VerseLink';
+import Skeleton from '@/dls/Skeleton/Skeleton';
+import useCurrentUser from '@/hooks/auth/useCurrentUser';
+import useGetReadingGoalProgress from '@/hooks/auth/useGetReadingGoalProgress';
+import { ReadingGoalType } from '@/types/auth/ReadingGoal';
+import { secondsFormatter } from '@/utils/datetime';
+import { toLocalizedNumber } from '@/utils/locale';
+
+const ReadingStreak = () => {
+  const { user, isLoading, error } = useCurrentUser();
+  const { t, lang } = useTranslation();
+  const { readingGoalProgress } = useGetReadingGoalProgress();
```

Comment:
Should we not handle error and loading states?

Reply:
Sorry, can you elaborate?

Reply:
`useGetReadingGoalProgress` is just a useSWR wrapper. What happens when the request fails or has `error`.